parasol/solr/client.py

```diff
@@ -0,0 +1,131 @@
+import inspect
+import json
+import logging
+import sys
+import time
+from urllib.parse import urljoin
+
+from attrdict import AttrDict
+import requests
+
+# Start making Django agnostic from the get-go
+try:
+    from django.settings import settings
+    from django.core.exceptions import ImproperlyConfigured
+except ImportError:
+    pass
+
+
+from parasol import __version__ as parasol_version
+from parasol.solr.schema import Schema
+from parasol.solr.update import Update
+from parasol.solr.admin import CoreAdmin
+
+logger = logging.getLogger(__name__)
+
+
+
+class SolrClientException(Exception):
+    '''Base class for all exceptions in this module'''
+    pass
+
+
+class CoreForTestExists(SolrClientException):
```

Comment:
Forgot to say - this seems like much too specific of an exception. How about just `CoreExists`, and the message would say the name of the core?